src/CoreWCF.Queue/tests/InMemoryQueue/InMemoryQueueTransportPump.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers;
+using System.Collections.Concurrent;
+using System.IO.Pipelines;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using CoreWCF.Queue.Common;
+
+namespace CoreWCF.Queue.Tests.InMemoryQueue;
+
+public class InMemoryQueueTransportPump : QueueTransportPump
+{
+    private readonly ConcurrentQueue<string> _queue;
+    private readonly ReceiveContextInterceptor _receiveContextInterceptor;
+    private CancellationTokenSource _cts;
+    private ManualResetEventSlim _mres;
+    public InMemoryQueueTransportPump(ConcurrentQueue<string> queue, ReceiveContextInterceptor receiveContextInterceptor)
+    {
+        _queue = queue;
+        _receiveContextInterceptor = receiveContextInterceptor;
+    }
+
+    public override Task StartPumpAsync(QueueTransportContext queueTransportContext, CancellationToken token)
+    {
+        _cts = new CancellationTokenSource();
+        _mres = new ManualResetEventSlim();
+        CancellationToken ct = CancellationTokenSource.CreateLinkedTokenSource(token, _cts.Token).Token;
+
```

Comment:
Linked cancelltion token sources should be disposed as they call token.Register on each of the passed tokens. This can create a memory leak.

Reply:
@mconnew I made the QueuePollingService (which is an IHostedService thus disposed by the host) disposable so we can support disposable QueueTransportPump implementation.